dali/test/python/test_operator_readers_webdataset_big.py

```diff
@@ -0,0 +1,101 @@
+from test_operator_readers_webdataset_base import *
+
+
+def cross_check(
+    dont_use_mmap,
+    batch_size,
+    num_shards,
+    shard_id,
+    prefetch_queue_depth,
+    skip_cached_images,
+    pad_last_batch,
+    lazy_init,
+    read_ahead,
+    stick_to_shard,
+):
+    num_multiplications = 5
+    num_samples = 100 * num_multiplications
+    tar_file_paths = [
+        get_dali_extra_path() + "/db/webdataset/sample-tar/cross.tar"
+    ] * num_multiplications
+    index_files = [generate_temp_index_file(tar_file_path) for tar_file_path in tar_file_paths]
+
+    extract_dirs = [generate_temp_extract(tar_file_path) for tar_file_path in tar_file_paths]
+    equivalent_files = sum(
+        (
+            sorted(
+                glob(extract_dir.name + "/*"),
+                key=lambda s: (int(s[s.rfind("/") + 1 : s.find(".")]), s),
+            )
+            for extract_dir in extract_dirs
+        ),
+        [],
+    )
+
+    compare_pipelines(
+        webdataset_raw_pipeline(
+            tar_file_paths,
+            [index_file.name for index_file in index_files],
+            ["a.a;a.b", "b.a;b.b"],
```

Comment:
Why we have `a.b` while `file_reader_pipeline` doesn't?

Reply:
since it is supposed to find the first extension, which in this case will be always a.a. Also a.b are set to be dirfiles, so they will be skipped anyways